### PR TITLE
[SPARK-40028][SQL][FollowUp] Improve examples of string functions

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Follow-up of https://github.com/apache/spark/pull/37458: improve the examples of string functions. Note that many of these functions already have binary example with input as `x'537061726b2053514c'`(binary of "Spark SQL"), so we don't need to make the example too complicated. 

### Why are the changes needed?

Make examples simpler.
### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

GA tests

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @beliefer

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala [hunk 1]
```diff
@@ -250,8 +250,8 @@ case class ConcatWs(children: Seq[Expression])
     Examples:
       > SELECT _FUNC_(1, 'scala', 'java');
        scala
-      > SELECT _FUNC_(1, encode('scala', 'utf-8'), encode('java', 'utf-8'));
-       scala
+      > SELECT _FUNC_(2, 'a', 1);
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:253]: note that ELT can accept any input type.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala [hunk 2]
```diff
@@ -539,16 +539,10 @@ case class BinaryPredicate(override val prettyName: String, left: Expression, ri
...
-       true
       > SELECT _FUNC_('Spark SQL', 'SPARK');
        false
-      > SELECT _FUNC_(encode('Spark SQL', 'utf-8'), encode('SPARK', 'utf-8'));
-       false
       > SELECT _FUNC_('Spark SQL', null);
        NULL
-      > SELECT _FUNC_(encode('Spark SQL', 'utf-8'), null);
-       NULL
       > SELECT _FUNC_(x'537061726b2053514c', x'537061726b');
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:546]: There is example for binary type already @beliefer

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala [hunk 3]
```diff
@@ -580,16 +574,10 @@ case class Contains(left: Expression, right: Expression) extends StringPredicate
...
-       true
       > SELECT _FUNC_('Spark SQL', 'SQL');
        false
-      > SELECT _FUNC_(encode('Spark SQL', 'utf-8'), encode('SQL', 'utf-8'));
-       false
       > SELECT _FUNC_('Spark SQL', null);
        NULL
-      > SELECT _FUNC_(encode('Spark SQL', 'utf-8'), null);
-       NULL
       > SELECT _FUNC_(x'537061726b2053514c', x'537061726b');
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:581]: There is example for binary type already @beliefer

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala [hunk 4]
```diff
@@ -1605,16 +1583,10 @@ case class BinaryPad(funcName: String, str: Expression, len: Expression, pad: Ex
...
-       hi???
       > SELECT _FUNC_('hi', 1, '??');
        h
-      > SELECT _FUNC_(encode('hi', 'utf-8'), 1, encode('??', 'utf-8'));
-       h
       > SELECT _FUNC_('hi', 5);
        hi
-      > SELECT _FUNC_(encode('hi', 'utf-8'), 5);
-       hi
       > SELECT hex(_FUNC_(unhex('aabb'), 5));
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:1590]: unhex returns binary results

**Review by HyukjinKwon — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala (hunk 1 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:253]: I see. Thank you.

**Review by beliefer — COMMENTED** [review]
`encode('Spark SQL ', 'utf-8')` have better readability than `x'537061726b2053514c'`.
Do we need to consider this factor ?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala (hunk 2 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:546]: I'm not familiar to the syntax x'537061726b2053514c'，I got it now.

**Comment by gengliangwang** [comment]
> encode('Spark SQL ', 'utf-8') have better readability than x'537061726b2053514c'.
Do we need to consider this factor ?

Well I am trying to follow the existing examples. Also, `x'537061726b2053514c'` is straightforward. For `encode('Spark SQL ', 'utf-8')`, users might need to look up the return type of `encode()` and wonder why encoding an ASCII string with utf-8.

**Comment by beliefer** [comment]
> > encode('Spark SQL ', 'utf-8') have better readability than x'537061726b2053514c'.
> > Do we need to consider this factor ?
> 
> Well I am trying to follow the existing examples. Also, `x'537061726b2053514c'` is straightforward. For `encode('Spark SQL ', 'utf-8')`, users might need to look up the return type of `encode()` and wonder why encoding an ASCII string with utf-8.

I got it.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala (hunk 1 above)
Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:253]: does this example involve type coercion?

**Comment by gengliangwang** [comment]
Merging to master

**Pull request closed without merging**